code/modules/martial_arts/sleeping_carp.dm

```diff
@@ -40,3 +40,8 @@
 
 /datum/martial_art/the_sleeping_carp/explaination_footer(user)
 	to_chat(user, "<b><i>In addition, by having your throw mode on when being shot at, you enter an active defense mode where you will block and deflect all projectiles fired at you!</i></b>")
+
+/datum/martial_art/the_sleeping_carp/try_deflect(mob/user)
+	if(user.in_throw_mode && prob(deflection_chance)) // in case an admin wants to var edit carp to have less deflection chance
+		return TRUE
+	return FALSE
```

Comment:
You basically did what I had in mind, just some streamlining now though.

In the base proc you can shorten this too:
```dm
/datum/martial_art/proc/try_deflect(mob/user)
	return prob(deflection_chance)
```

And then here you can do:

```dm
/datum/martial_art/the_sleeping_carp/try_deflect(mob/user)
	return ..() && user.in_throw_mode // in case an admin wants to var edit carp to have less deflection chance
```
Which takes advantage of the parent proc.